File: templates/backend/express-mysql/src/routes/posts.ts
import { Router } from 'express';
import { pool } from '../config/database.js';
import { authenticate } from '../middleware/authenticate.js';
import { AppError } from '../middleware/errorHandler.js';

const router = Router();

router.get('/', authenticate, async (req, res, next) => {
  try {
    const [posts] = await pool.query('SELECT * FROM posts WHERE author_id = ? ORDER BY created_at DESC', [req.user!.userId]);
    res.json({ success: true, data: posts });
  } catch (err) {
    next(err);
  }
});

router.post('/', authenticate, async (req, res, next) => {
  try {
    const { title, content, published } = req.body;
    const [result] = await pool.query('INSERT INTO posts (title, content, published, author_id) VALUES (?, ?, ?, ?)', [title, content, published || false, req.user!.userId]);
    res.status(201).json({ success: true, data: { id: (result as any).insertId, title, content, published } });
  } catch (err) {
    next(err);
  }
});

router.put('/:id', authenticate, async (req, res, next) => {
  try {
    const [result] = await pool.query('UPDATE posts SET ? WHERE id = ? AND author_id = ?', [req.body, req.params.id, req.user!.userId]);
    if (!(result as any).affectedRows) throw new AppError('Post not found', 404);
    res.json({ success: true, message: 'Post updated' });
  } catch (err) {
    next(err);
  }
});

router.delete('/:id', authenticate, async (req, res, next) => {
  try {
    const [result] = await pool.query('DELETE FROM posts WHERE id = ? AND author_id = ?', [req.params.id, req.user!.userId]);
    if (!(result as any).affectedRows) throw new AppError('Post not found', 404);
    res.json({ success: true, message: 'Post deleted' });
  } catch (err) {
    next(err);
  }
});

export default router;

File: templates/backend/flask-sqlite/src/routes/posts.py
from flask import Blueprint, request, jsonify
from flask_jwt_extended import jwt_required, get_jwt_identity
from ..config.database import db
from ..models.models import Post

bp = Blueprint('posts', __name__)

@bp.route('/', methods=['GET'])
@jwt_required()
def get_posts():
    user_id = get_jwt_identity()
    posts = Post.query.filter_by(user_id=user_id).all()
    return jsonify({'success': True, 'data': [{'id': p.id, 'title': p.title, 'content': p.content, 'published': p.published} for p in posts]})

@bp.route('/', methods=['POST'])
@jwt_required()
def create_post():
    user_id = get_jwt_identity()
    data = request.json
    post = Post(title=data['title'], content=data.get('content'), published=data.get('published', False), user_id=user_id)
    db.session.add(post)
    db.session.commit()
    return jsonify({'success': True, 'data': {'id': post.id, 'title': post.title}}), 201

@bp.route('/<int:post_id>', methods=['PUT'])
@jwt_required()
def update_post(post_id):
    user_id = get_jwt_identity()
    post = Post.query.filter_by(id=post_id, user_id=user_id).first()
    if not post:
        return jsonify({'error': 'Post not found'}), 404
    data = request.json
    post.title = data.get('title', post.title)
    post.content = data.get('content', post.content)
    post.published = data.get('published', post.published)
    db.session.commit()
    return jsonify({'success': True, 'message': 'Post updated'})

@bp.route('/<int:post_id>', methods=['DELETE'])
@jwt_required()
def delete_post(post_id):
    user_id = get_jwt_identity()
    post = Post.query.filter_by(id=post_id, user_id=user_id).first()
    if not post:
        return jsonify({'error': 'Post not found'}), 404
    db.session.delete(post)
    db.session.commit()
    return jsonify({'success': True, 'message': 'Post deleted'})
